Sources/DDGSync/SyncFeatureFlags.swift

```diff
@@ -68,20 +69,39 @@ public struct SyncFeatureFlags: OptionSet {
     // MARK: -
 
     init(privacyConfig: PrivacyConfiguration) {
-        guard privacyConfig.isEnabled(featureKey: .sync) else {
+        var offFlag: SyncSubfeature?
+        let syncState = privacyConfig.stateFor(featureKey: .sync)
+        switch syncState {
+
+        case .enabled:
+            if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level0ShowSync) {
+                offFlag = .level0ShowSync
+                self = .unavailable
+            } else if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level1AllowDataSyncing) {
+                offFlag = .level1AllowDataSyncing
+                self = .level0ShowSync
+            } else if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level2AllowSetupFlows) {
+                offFlag = .level2AllowSetupFlows
+                self = .level1AllowDataSyncing
+            } else if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level3AllowCreateAccount) {
+                offFlag = SyncSubfeature.level3AllowCreateAccount
+                self = .level2AllowSetupFlows
+            } else {
+                self = .level3AllowCreateAccount
+            }
+        case .disabled(let reason):
+            unavailableReason = reason
             self = .unavailable
-            return
         }
-        if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level0ShowSync) {
-            self = .unavailable
-        } else if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level1AllowDataSyncing) {
-            self = .level0ShowSync
-        } else if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level2AllowSetupFlows) {
-            self = .level1AllowDataSyncing
-        } else if !privacyConfig.isSubfeatureEnabled(SyncSubfeature.level3AllowCreateAccount) {
-            self = .level2AllowSetupFlows
-        } else {
-            self = .level3AllowCreateAccount
+
+        if let offFlag {
+            let state = privacyConfig.stateFor(offFlag)
+            switch state {
+            case .enabled:
+                break
+            case .disabled(let reason):
+                unavailableReason = reason
+            }
```

Comment:
This was a bit confusing to me at the beginning and I had to follow the code closely to understand it better. Could we rewrite it as follows?

```suggestion
        if let offFlag, case .disabled(let reason) = privacyConfig.stateFor(offFlag) {
            unavailableReason = reason
        }
```

Reply:
I personally find this less readable because is condensed but I’ll change it… cause it's probably just me 😄